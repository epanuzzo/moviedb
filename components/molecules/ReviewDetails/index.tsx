import React from "react";

export type ReviewDetailsProps = {
  user: string;
  rating: number;
  comment: string;
  date: string;
};

/**
 * ReviewDetails component displays the details of a movie review.
 *
 * @component
 * @param {ReviewDetailsProps} props - The properties object.
 * @param {string} props.user - The name of the user who wrote the review.
 * @param {number} props.rating - The rating given by the user, out of 10.
 * @param {string} props.comment - The comment provided by the user.
 * @param {string} props.date - The date when the review was written.
 * @returns {JSX.Element} The rendered component displaying the review details.
 */
const ReviewDetails: React.FC<ReviewDetailsProps> = ({
  user,
  rating,
  comment,
  date,
}) => {
  const formatedDate = new Date(date).toLocaleDateString();
  return (
    <div className="bg-gray-800 p-4 rounded-lg mb-4">
      <div className="flex justify-between">
        <p className="font-semibold">{user}</p>
        <p className="text-sm">{rating} / 10</p>
      </div>
      <p className="text-gray-400 mt-2">{comment}</p>
      <p className="text-xs text-gray-500 mt-2">{formatedDate}</p>
    </div>
  );
};

export default ReviewDetails;
