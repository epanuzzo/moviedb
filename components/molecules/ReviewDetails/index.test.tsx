import React from "react";
import renderer from "react-test-renderer";
import ReviewDetails from "./index";

describe("ReviewDetails", () => {
  it("renders component", () => {
    const tree = renderer
      .create(
        <ReviewDetails
          user="User"
          rating={5}
          comment="Comment"
          date="2021-01-01"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
