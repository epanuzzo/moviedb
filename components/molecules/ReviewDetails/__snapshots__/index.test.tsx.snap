// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewDetails renders component 1`] = `
<div
  className="bg-gray-800 p-4 rounded-lg mb-4"
>
  <div
    className="flex justify-between"
  >
    <p
      className="font-semibold"
    >
      User
    </p>
    <p
      className="text-sm"
    >
      5
       / 10
    </p>
  </div>
  <p
    className="text-gray-400 mt-2"
  >
    Comment
  </p>
  <p
    className="text-xs text-gray-500 mt-2"
  >
    01/01/2021
  </p>
</div>
`;
